### PR TITLE
Sort Devices and Groups in Alert Rules 'map to' droplist

### PR DESCRIPTION
**Pull request by SourceDoctor**
sort Devices and Groups alphabetic in
Alert Rule -> "Map To"

DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [X] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 10530 `
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by PipoCanaja** [comment]
Works well for the groups. Not so OK for devices, because they depend on force_ip_to_sysname option. 
You need to sort in PHP (and not in the SQL part) the $devices array, in which the text was already processed by format_hostname according to user preferences.

**Comment by SourceDoctor** [comment]
@PipoCanaja 
sorting is moved

File: includes/html/list/devices.inc.php [hunk 1]
```diff
@@ -68,4 +68,6 @@
 
 $more = ($offset + count($devices)) < $total;
 
+array_multisort(array_column($devices, 'text'), SORT_ASC, $devices);
```
Comment (laf) [inline includes/html/list/devices.inc.php]: `text` isn't a valid column name in the `devices` table.

Also, you need to check if `Config::get('force_ip_to_sysname')` is true and use `sysName` for sorting if not, `hostname` should be used.

Comment (SourceDoctor) [reply includes/html/list/devices.inc.php]: @laf 
"format_hostname()" returns the hostname string, and checks also 'force_ip_to_sysname'

this function will be used in an array_mapping function which returns an array of associative arrays, containing the keys 'id' and 'text'. 'text' contains the return value of "format_hostname()"
Take a look on lin 61-67

so on my views there is nothing to change
(i tested the code and it works well)

Comment (PipoCanaja) [reply includes/html/list/devices.inc.php]: Good for me, tested all right

**Comment by murrant** [comment]
This pull request has been mentioned on **LibreNMS Community**. There might be relevant details there:

https://community.librenms.org/t/v1-55-release-changelog-august-2019/9428/1

**Pull request merged**